src/rgw/rgw_rados.cc

```diff
@@ -12892,25 +13048,36 @@ int RGWRados::cls_obj_set_bucket_tag_timeout(RGWBucketInfo& bucket_info, uint64_
   return CLSRGWIssueSetTagTimeout(index_ctx, bucket_objs, cct->_conf->rgw_bucket_index_max_aio, timeout)();
 }
 
-int RGWRados::cls_bucket_list(RGWBucketInfo& bucket_info, int shard_id, rgw_obj_index_key& start, const string& prefix,
-		              uint32_t num_entries, bool list_versions, map<string, rgw_bucket_dir_entry>& m,
-			      bool *is_truncated, rgw_obj_index_key *last_entry,
-			      bool (*force_check_filter)(const string&  name))
+
+int RGWRados::cls_bucket_list_ordered(RGWBucketInfo& bucket_info,
+				      int shard_id,
+				      rgw_obj_index_key& start,
+				      const string& prefix,
+				      uint32_t num_entries,
+				      bool list_versions,
+				      map<string, rgw_bucket_dir_entry>& m,
+				      bool *is_truncated,
+				      rgw_obj_index_key *last_entry,
+				      bool (*force_check_filter)(const string& name))
 {
-  ldout(cct, 10) << "cls_bucket_list " << bucket_info.bucket << " start " << start.name << "[" << start.instance << "] num_entries " << num_entries << dendl;
+  ldout(cct, 10) << "cls_bucket_list " << bucket_info.bucket << " start " <<
```

Comment:
cls_bucket_list_ordered is better, right?

Reply:
Yes! And the logging output later in the function as well!